pandas/core/dtypes/cast.py

```diff
@@ -1225,6 +1225,9 @@ def maybe_cast_to_datetime(
     _ensure_nanosecond_dtype(dtype)
 
     if lib.is_np_dtype(dtype, "m"):
+        if getattr(value, "ndim", 1) == 2 and value.shape[1] == 1:
+            res = TimedeltaArray._from_sequence(value.ravel(), dtype=dtype)
+            return res.reshape(value.shape)
         res = TimedeltaArray._from_sequence(value, dtype=dtype)
         return res
```

Comment:
Should we raise error here for other situations, e.g. when the shape is not (N, 1) or (1, )?

Or is a less strict check appropriate, like `if getattr(value, "ndim", 1) > 1:`? @rhshadrach

Reply:
I don't think we should ever be reaching here with ndim 3 or higher. E.g.

```python
arr = np.zeros((3, 3, 3))
pd.DataFrame(arr)
# ValueError: Must pass 2-d input. shape=(3, 3, 3)
```